src/consensus/aft/test/logging_stub.h

```diff
@@ -72,15 +69,52 @@ namespace aft
   {
   public:
     // Capture what is being sent out
-    std::list<std::pair<ccf::NodeId, RequestVote>> sent_request_vote;
-    std::list<std::pair<ccf::NodeId, AppendEntries>> sent_append_entries;
-    std::list<std::pair<ccf::NodeId, RequestVoteResponse>>
-      sent_request_vote_response;
-    std::list<std::pair<ccf::NodeId, AppendEntriesResponse>>
-      sent_append_entries_response;
+    // Using a deque so we can both pop from the front and shuffle
+    using MessageList =
+      std::deque<std::pair<ccf::NodeId, std::vector<uint8_t>>>;
+    MessageList messages;
```

Comment:
Collapsing these separate queues into a single combined queue, so its easier to shuffle/drop arbitrary messages. Means we need to peek at the serialised form to rediscover the message type in more places if we're looking for something specific, but I think this is worth it.